mbbsd/bbs.c

```diff
@@ -3690,6 +3711,32 @@ mark_post(int ent, fileheader_t * fhdr, const char *direct)
     return change_post_mode(ent, fhdr, direct, FILE_MARKED);
 }
 
+static int
+mark_suspicious(int ent, fileheader_t * fhdr, const char *direct)
+{
+    boardheader_t *bp = getbcache(currbid);
+
+    aidu_t aidu = 0;
+    aidu = fn2aidu((char *)fhdr->filename);
+    if (aidu > 0) {
+        char aidc[10];
+        char buf[256];
+        aidu2aidc(aidc, aidu);
+
+        if (fhdr->filemode & FILE_SUSPICIOUS)
+            snprintf(buf, sizeof(buf),
+                     "%s �N #%s (%s) �Ѱ��ݬd��", cuser.userid, aidc, bp->brdname);
+        else
+            snprintf(buf, sizeof(buf),
+                     "%s �N #%s (%s) �аO�ݬd��", cuser.userid, aidc, bp->brdname);
+
+        post_msg(BN_SECURITY, buf, "�Ъ`�N�аO���X�k��", "[���ҼаO]");
```

Comment:
The user needs to pass the permission check (`MODE_BOARD`) before they can do this. And this is done in `change_post_mode`. 
This will cause erroneous logging for normal users.

Reply:
Because ```mark_suspicious``` was moved into ```manage_post``` and the permission check is done at the beginning of ```manage_post```, I think there is no need to check the permission in ```mark_suspicious``` redundantly.